src/parser/mod.rs

```diff
@@ -2957,6 +2957,15 @@ impl<'a> Parser<'a> {
         self.expect_keyword(Keyword::AS)?;
         let query = Box::new(self.parse_query()?);
         // Optional `WITH [ CASCADED | LOCAL ] CHECK OPTION` is widely supported here.
+
+        let late_binding = dialect_of!(self is RedshiftSqlDialect)
```

Comment:
Can you please also parse this syntax with GenericDialect as well?

```suggestion
        let late_binding = dialect_of!(self is RedshiftSqlDialect | GenericDialect)
```